jvb/src/main/kotlin/org/jitsi/videobridge/Endpoint.kt

```diff
@@ -837,17 +887,71 @@ class Endpoint @JvmOverloads constructor(
         bitrateController.setBandwidthAllocationSettings(message)
     }
 
+    /**
+     * Find the properties of the video source indicated by the given SSRC. Returns null if not found.
+     */
+    private fun findVideoSourceProps(ssrc: Long): MediaSourceDesc? {
+        conference.getEndpointBySsrc(ssrc)?.let { ep ->
+            ep.mediaSources.forEach { s ->
+                if (s.findRtpEncodingDesc(ssrc) != null) {
+                    return s
+                }
+            }
+        }
+        logger.error { "\tNo properties found for SSRC $ssrc." }
+        return null
+    }
+
+    /**
+     * Find the properties of the audio source indicated by the given SSRC. Returns null if not found.
+     */
+    private fun findAudioSourceProps(ssrc: Long): AudioSourceDesc? {
+        conference.getEndpointBySsrc(ssrc)?.let { ep ->
+            if (ep !is Endpoint)
+                return null
+            return ep.audioSources.find { s -> s.ssrc == ssrc }
+        }
+        logger.error { "\tNo properties found for SSRC $ssrc." }
+        return null
+    }
+
+    /**
+     * Find the properties of the source indicated by the given SSRC. Returns null if not found.
+     */
+    private fun findSourceProps(ssrc: Long, mediaType: MediaType): SourceDesc? {
+        if (mediaType == MediaType.AUDIO) {
+            val p = findAudioSourceProps(ssrc)
+            if (p == null || p.sourceName == null || p.owner == null)
+                return null
+            else
+                return SourceDesc(p)
+        } else {
+            val p = findVideoSourceProps(ssrc)
+            if (p == null || p.sourceName == null || p.owner == null)
+                return null
+            else
+                return SourceDesc(p)
+        }
+    }
+
     override fun send(packetInfo: PacketInfo) {
         when (val packet = packetInfo.packet) {
             is VideoRtpPacket -> {
                 if (bitrateController.transformRtp(packetInfo)) {
                     // The original packet was transformed in place.
+                    if (doSsrcRewriting) {
+                        val start = packet !is ParsedVideoPacket || (packet.isKeyframe && packet.isStartOfFrame)
+                        if (!videoSsrcs.rewriteRtp(packet, start)) {
+                            return
+                        }
+                    }
                     transceiver.sendPacket(packetInfo)
                 } else {
                     logger.warn("Dropping a packet which was supposed to be accepted:$packet")
                 }
                 return
             }
+            is AudioRtpPacket -> if (doSsrcRewriting) audioSsrcs.rewriteRtp(packet)
             is RtcpSrPacket -> {
```

Comment:
You're not doing any rewriting of RTCP packets - would this be needed? It seems like the receivers will get RTCP for all the original sources, and none for the re-written ones.  I don't know if this matters in practice for the browsers, but it somewhat concerns me in theory?

Reply:
completed.